sidebar/js/components/Sidebar.tsx

```diff
@@ -0,0 +1,274 @@
+import React, { useState, useEffect } from "react";
+
+export interface Data {
+  sections: Section[];
+  page: number;
+  previous_page: null;
+  next_page: null;
+  per_page: number;
+  page_count: number;
+  count: number;
+  sort_by: string;
+  sort_order: string;
+  categories: Category[];
+}
+
+export interface Category {
+  id: number;
+  url: string;
+  html_url: string;
+  position: number;
+  created_at: Date;
+  updated_at: Date;
+  name: string;
+  description: string;
+  locale: Locale;
+  source_locale: Locale;
+  outdated: boolean;
+}
+
+export enum Locale {
+  EnUs = "en-us"
+}
+
+export interface Section {
+  id: number;
+  url: string;
+  html_url: string;
+  category_id: number;
+  position: number;
+  sorting: Sorting;
+  created_at: Date;
+  updated_at: Date;
+  name: string;
+  description: string;
+  locale: Locale;
+  source_locale: Locale;
+  outdated: boolean;
+  parent_section_id: number | null;
+  theme_template: ThemeTemplate;
+}
+
+export enum Sorting {
+  Manual = "manual",
+  Title = "title"
+}
+
+export enum ThemeTemplate {
+  SectionPage = "section_page",
+  SectionPagesChatPage = "section_pages/chat_page"
+}
+
+interface Props {
+  // categories: Category[];
+}
+
+var category1 = [
+  {
+    name: "CET Designer",
+    sections: [
+      {
+        name: "One",
+        id: 1,
+        href: "/"
+      },
+      {
+        name: "Two",
+        id: 2,
+        href: "/"
+      }
+    ]
+  },
+  {
+    name: "Catalogues",
+    sections: [
+      {
+        name: "One",
+        id: 21,
+        href: "/"
+      },
+      {
+        name: "Two",
+        id: 22,
+        href: "/"
+      }
+    ]
+  },
+  {
+    name: "MyConfigura",
+    sections: [
+      {
+        name: "One",
+        id: 21,
+        href: "/"
+      },
+      {
+        name: "One",
+        id: 21,
+        href: "/"
+      },
+      {
+        name: "One",
+        id: 21,
+        href: "/"
+      },
+      {
+        name: "One",
+        id: 21,
+        href: "/"
+      },
+      {
+        name: "One",
+        id: 21,
+        href: "/"
+      },
+      {
+        name: "Two",
+        id: 22,
+        href: "/"
+      }
+    ]
+  }
+];
+
+export default function Sidebar(props: Props) {
+  // const [open, setOpen] = useState(false);
+  const [openId, setOpenId] = useState(0);
+  const [data, setData] = useState<Data>();
+  const [liClass, setListClass] = useState("sidebar-item");
+  const url =
+    "/api/v2/help_center/en-us/sections.json?include=categories&per_page=100";
+
+  function expand(id: number) {
+    setOpenId(prevState => {
+      if (prevState === id) {
+        return 0;
+      }
+      return id;
+    });
+    // setOpen(!open);
+  }
+
+  async function fetchUrl() {
```

Comment:
this doesn't need to be an nextra function, the code can just be embedded in the useEffect 🙂